cmd/resolve.go

```diff
@@ -0,0 +1,672 @@
+				oldOp := config.Operator
+				oldAc := config.Account
+				v, ok := q["store"]
+				if ok {
+					defer func() {
+						_ = config.setStoreRoot(oldSR)
+						if oldOp != "" {
+							_ = config.SetOperator(oldOp)
+						}
+						if oldAc != "" {
+							_ = config.SetAccount(oldAc)
+						}
+					}()
+					sr, err := Expand(v)
+					if err != nil {
+						return err
+					}
+					if err := config.setStoreRoot(sr); err != nil {
+						return err
+					}
+					if err := config.SetOperator(u.operator); err != nil {
+						return err
+					}
+				}
+
+				storeDir, ok := q[keystoreDir]
+				if ok {
+					ks, err := Expand(storeDir)
+					if err != nil {
+						return err
+					}
+					if err := os.Setenv(store.NKeysPathEnv, ks); err != nil {
+						return err
+					}
+				}
+			}
+			return RunAction(cmd, args, &params)
+		},
+		Hidden: true,
+	}
+	cmd.Flags().StringVarP(&params.outputFile, "output-file", "o", "--", "output file, '--' is stdout")
+
+	return cmd
+}
+
+func init() {
+	generateCmd.AddCommand(createProfileCmd())
+}
+
+type Arg string
+
+const (
+	prefix           = "nsc://"
+	operatorKey  Arg = "operatorkey"
+	accountKey   Arg = "accountkey"
+	userKey      Arg = "userkey"
+	key          Arg = "key"
+	operatorSeed Arg = "operatorseed"
+	accountSeed  Arg = "accountseed"
+	userSeed     Arg = "userseed"
+	seed         Arg = "seed"
+	operatorName Arg = "operatorname"
+	accountName  Arg = "accountname"
+	userName     Arg = "username"
+	name         Arg = "name"
+	keystoreDir  Arg = "keystore"
+	storeDir     Arg = "store"
+)
+
+type ProfileCmdParams struct {
+	nscu       *NscURL
+	results    *Profile
+	oc         *jwt.OperatorClaims
+	ac         *jwt.AccountClaims
+	uc         *jwt.UserClaims
+	outputFile string
+}
+
+type Details struct {
+	Service []string `json:"service,omitempty"`
+	Name    string   `json:"name,omitempty"`
+	Seed    string   `json:"seed,omitempty"`
+	Key     string   `json:"id,omitempty"`
+}
+
+type Profile struct {
+	UserCreds string   `json:"user_creds,omitempty"`
+	Operator  *Details `json:"operator,omitempty"`
+	Account   *Details `json:"account,omitempty"`
+	User      *Details `json:"user,omitempty"`
+}
+
+type NscURL struct {
+	operator string
+	account  string
+	user     string
+	qs       string
+}
+
+func (u *NscURL) getOperator() (string, error) {
+	return url.QueryUnescape(u.operator)
+}
+
+func (u *NscURL) getAccount() (string, error) {
+	return url.QueryUnescape(u.account)
+}
+
+func (u *NscURL) getUser() (string, error) {
+	return url.QueryUnescape(u.user)
+}
+
+func (u *NscURL) query() (map[Arg]string, error) {
+	q := strings.ToLower(u.qs)
+	m := make(map[Arg]string)
+	for _, e := range strings.Split(q, "&") {
+		kv := strings.Split(e, "=")
+		k := strings.ToLower(kv[0])
+		v := ""
+		if len(kv) == 2 {
+			s, err := url.QueryUnescape(kv[1])
+			if err != nil {
+				return nil, err
+			}
+			v = s
+		}
+		m[Arg(k)] = v
+	}
+	return m, nil
+}
+
+func ParseNscURL(u string) (*NscURL, error) {
```

Comment:
I assume this id done manually as we don't use a host and so url parsing may be tricky?

Reply:
correct - I am allowing spaces or URL encoded.